windows/installer.nsi

```diff
@@ -212,6 +218,30 @@ Section "Install_64Bit" INST_64
                 MessageBox MB_OKCANCEL|MB_ICONEXCLAMATION "$(OLD_VER_REMOVE_FAILED)" /sd IDCANCEL IDOK continueinstall
                 goto quit
             ${EndIf}
+
+            ; ask to move old ASIO4ALL registry configuration
+
+            ; check if old ASIO4ALL keys exit
+
+            ClearErrors
+            EnumRegKey $0 HKCU "SOFTWARE\ASIO4ALL v2 by Wuschel\8A9E7A56" 0
+
+            IfErrors 0 continueinstall
+                MessageBox MB_YESNO|MB_ICONEXCLAMATION "Do you want to copy ASIO4ALL settings from the previous version if possible?" /sd IDYES IDNO continueinstall
+
+                    IntOp $0 0 + 0
+                    EnumStart:
```

Comment:
For each user, replace the old with the new key contents

Reply:
Should we do that? It feels wrong to update other users and not just HKCU.

Reply:
On my PC I have two users: 
1. Admin (user which I use to install Jamulus)
2. User (user which I use to use Jamulus)
If we don't  copy all users' settings, this issue would only be fixed for Admin but not for User.

Reply:
But if Admin has never run Jamulus, it won't find the ASIO key in HKCU? Or if it does, the settings may well not be the same as were set the last time the User user Jamulus? Does the check for existing settings also need to scan the registry for users then?

The installer presumably has admin privileges, because I've never switched to admin to install Jamulus.

Reply:
> But if Admin has never run Jamulus, it won't find the ASIO key in HKCU? 

On my PC at least, Admin did run Jamulus. 

> Does the check for existing settings also need to scan the registry for users then?

Yes. This makes this code a lot more complicated.

Furthermore something like that would have to be re-implemented (but it's not that easy unfortunately):

foreach (user in users) {
    if (user.reg.oldkey.exists() && !user.reg.newkey.exists()) {
        user.reg.newkey.setValue(oldkey.getValue());
    }
}

Reply:
This pseudo-code sounds how it should work. Too bad that NSIS seems to be so complicated. :(

If NSIS implementation is the problem, would there be other workarounds? E.g. replacing the Jamulus shortcut by one pointing to a wrapper script which performs the per-user migration (maybe in Powershell?) and launches Jamulus.exe afterwards?
Not sure what side effects we would have to check for then or how it would look like to the user.

Reply:
NSIS should support that. We could use https://nsis.sourceforge.io/mediawiki/index.php?title=Registry_plug-in&oldid=16847 but I don't really want to include another plug-in. We already have too many in the installer.

Reply:
Ok. I think I got it to work now.